.ci/Jenkinsfile

```diff
@@ -244,33 +253,94 @@ pipeline {
+                 ${env.WORKSPACE}/.venv/bin/pip3 -q install wheel
+                 ${env.WORKSPACE}/.venv/bin/pip3 -q install ansible requests google-auth boto3 boto
+                 ${env.WORKSPACE}/.venv/bin/ansible-galaxy install geerlingguy.docker
+            fi
+            ${env.WORKSPACE}/.venv/bin/${command}
+          """,
+           label: "Executing pyrun"
+        )
       }
-      def pullRequestFilter = scenario.containsKey('pullRequestFilter') ? scenario.pullRequestFilter : ''
-      def tags = scenario.tags
-      def regexps = [ "^e2e/_suites/${suite}/.*", "^.ci/.*", "^cli/.*", "^e2e/.*\\.go", "^internal/.*\\.go" ]
-      if ("${FORCE_SKIP_GIT_CHECKS}" == "true" || isGitRegionMatch(patterns: regexps, shouldMatchAll: false)) {
-        platformsValue.each { platform ->
-          def platformLabels = platform.labels
+    }
+  }
+}
+
+def ansible(args){
+     pyrun("ansible-playbook -v --private-key=\"${env.WORKSPACE}/${env.BASE_DIR}/e2essh\" --extra-vars=\"workspace=${env.WORKSPACE}/${env.BASE_DIR} numAgents=4 runId=${env.RUN_ID.split('-')[0]} sshPublicKey=${env.WORKSPACE}/${env.BASE_DIR}/e2essh.pub\" --ssh-common-args='-o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null' ${env.WORKSPACE}/${env.BASE_DIR}/.ci/ansible/playbook.yml ${args}")
+}
+
+
+def checkTestSuite(Map parallelTasks = [:], Map item = [:]) {
+  def suite = item.suite
+  def platforms = item.platforms
+
+  if(suite == "fleet"){
+    // Deploy the test infrastructure to fleet related tests
+    sh "ssh-keygen -b 4096 -t rsa -f ${env.WORKSPACE}/${env.BASE_DIR}/e2essh -q -N \"\" "
+    ansible("-t provision-stack")
+  }
+
+  item.scenarios.eachWithIndex { scenario, scenarioIdx ->
+    def name = scenario.name
+    def platformsValue = platforms
+    def scenarioPlatforms = scenario.platforms
+    if (scenarioPlatforms?.size() > 0) {
+      // scenario platforms take precedence over suite platforms, overriding them
+      platformsValue = scenarioPlatforms
+    }
+    def pullRequestFilter = scenario.containsKey('pullRequestFilter') ? scenario.pullRequestFilter : ''
+    def tags = scenario.tags
+    def regexps = [ "^e2e/_suites/${suite}/.*", "^.ci/.*", "^cli/.*", "^e2e/.*\\.go", "^internal/.*\\.go" ]
+    if ("${FORCE_SKIP_GIT_CHECKS}" == "true" || isGitRegionMatch(patterns: regexps, shouldMatchAll: false)) {
+      platformsValue.each { platform ->
+        def platformLabels = platform.labels
+        if(suite == "fleet" && (platformLabels == "arm64" || platformLabels == "amd64")){
+          def testRunners = readYaml(file: "${env.WORKSPACE}/${env.BASE_DIR}/${platformLabels}-sshhosts")
+          log(level: 'INFO', text: "Adding ${platformLabels}:${suite}:${tags} test suite to the build execution")
+          parallelTasks["${platformLabels}_${suite}_${tags}"] = generateFunctionalTestStep(name: "${name}", platform: platform,  suite: "${suite}", tags: "${tags}", pullRequestFilter: "${pullRequestFilter}", testRunner: testRunners[scenarioIdx])
```

Comment:
I guess this is going to spread the test workloads into one host, defined by `scenarioIdx`:

First scenario to the first host, and so on, right? If that's the case, we will be creating as many hosts as items in the YAML descriptor